### ISSUE TITLE
Modal with Form on:submit method

### ISSUE THREAD

**Issue by albertms10** [carbon-design-system/carbon-components-svelte#310]
When a Form is placed inside a Modal, the Modal `on:submit` method is called with a `target: null` property either clicking on the primary button or hitting enter (as `shouldSubmitOnEnter` is `true` by default). Shouldn’t it contain a reference to the `HTMLFormElement` instead? Or do I need to manually manage `on:click` primary button to submit the form?

```svelte
<Modal
  ...
  hasForm
  on:submit={(e) => {
    console.log('modal', e);
  }}
>
  <Form
    on:submit={(e) => {
      console.log('form', e); // Is never printed
    }}
  >
    <TextInput name="name" labelText="Name" required />
    <TextInput name="email" labelText="Email" required />
  </Form>
</Modal>
```

```console
modal CustomEvent {..., target: null}
```

**Comment by metonym** [comment]
This is most definitely a bug. I'd expect `<Modal on:submit />` to return a reference to the form. Would you like to work on this?

**Comment by albertms10** [comment]
I don’t know if I understand what should be going on here, but, for the Modal `on:submit` event to contain a reference to the Form as a target, it should be placed inside the Modal component definition, not as a child of it, right?

So, @metonym, what is the expected way of this to work in an _automatic_ way?—without having to manually manage the Form reference to the Modal outside of it.

**Comment by metonym** [comment]
After looking at this again, I don't think the Form state should be managed by Modal. That being said, I feel that a prop like `hasForm` ideally should be set automatically if a Form is present. `hasForm` is purely cosmetic.

**Comment by albertms10** [comment]
As Svelte cannot know which elements are contained in the component’s `slot`, `hasForm` could be there to set the appropriate style—or the CSS styles should take that into account.
So, as you stated, the Modal `on:submit` should not be considered to also submit the Form content. To make things easy, something like what is described [in this REPL](https://svelte.dev/repl/53b51d2080914e71bb1fbaf78d65d444?version=3.29.4)—by using stores—should make it work.
If this is the expected behaviour, we should add an example to the docs, as well.

**Comment by brunnerh** [comment]
Both `hasForm` and `hasScrollingContent` seem lacking in terms of documentation.

**Comment by sallaben** [comment]
`shouldSubmitOnEnter` doesn't seem to work at all!

**Comment by metonym** [comment]
@sallaben Good catch. Tracking in a separate issue: https://github.com/carbon-design-system/carbon-components-svelte/issues/1005

**Comment by WilliamDiakite** [comment]
The solution I've used so far is to pass the id of the form element to the Modal. Because the button set of the modal is outside the form, clicking on the primary button wont trigger a sumbit on the form. Adding  `type="submit"` and a `form` attribute to the primary button with value of `formId` does the trick. 

Here is a [REPL](https://svelte.dev/repl/9c2b5739fea54318b1450bb158caf8a4?version=3.48.0) illustrating the suggested solution.

I would be happy to make a pull request to fix this bug asap.

**Comment by metonym** [comment]
@WilliamDiakite A PR would be appreciated

**Comment by metonym** [comment]
Fixed in [v0.95.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.95.0).

`Modal` accepts a `formId` prop to associate with a native `form` element. [Example.](https://svelte.carbondesignsystem.com/components/Modal#with-form-submission)

```svelte
<script>
  import { Button, Modal, Stack, TextInput } from "carbon-components-svelte";

  let open = false;
  let name = "";
  let email = "";

  function handleSubmit() {
    console.log("Form submitted:", { name, email });
    open = false;
  }
</script>

<Button on:click={() => (open = true)}>Create account</Button>

<Modal
  bind:open
  modalHeading="Create account"
  primaryButtonText="Submit"
  secondaryButtonText="Cancel"
  hasForm
  formId="account-form"
  on:click:button--secondary={() => (open = false)}
>
  <form id="account-form" on:submit|preventDefault={handleSubmit}>
    <Stack gap={5}>
      <TextInput bind:value={name} labelText="Name" placeholder="Enter name" />
      <TextInput
        bind:value={email}
        labelText="Email"
        type="email"
        placeholder="Enter email"
      />
    </Stack>
  </form>
</Modal>

```
